lib/ansible/modules/cloud/azure/azure_rm_route.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_route
+version_added: "2.7"
+short_description: Manage Azure route resource.
+description:
+    - Create, update or delete a route.
+options:
+    resource_group:
+        description:
+            - name of resource group.
+        required: true
+    name:
+        description:
+            - name of the virtual network.
+        required: true
+    state:
+        description:
+            - Assert the state of the virtual network. Use 'present' to create or update and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+    address_prefix:
+        description:
+            - The destination CIDR to which the route applies.
+    next_hop_type:
+        description:
+            - The type of Azure hop the packet should be sent to.
+        choices:
+            - VirtualNetworkGateway
+            - VnetLocal
+            - Internet
+            - VirtualAppliance
+            - None
```

Comment:
Using "None" is a kind of weird, as it may get confused with python's None
Maybe there should be no choice "None" at all?

Reply:
Since this field  is required to identify the type when creation. Here I would like to make the default value is `str(None)`. Then user can use `None` without setting.